src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,287 @@
 package internet.shop.dao.impl;
 
 import internet.shop.dao.CarDao;
-import internet.shop.db.Storage;
+import internet.shop.exception.DataProcessingException;
 import internet.shop.lib.Dao;
 import internet.shop.model.Car;
+import internet.shop.model.Driver;
+import internet.shop.model.Manufacturer;
+import internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Optional;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 
 @Dao
 public class CarDaoImpl implements CarDao {
     @Override
     public Car create(Car car) {
-        Storage.addCar(car);
+        String queryForCar = "INSERT INTO cars (manufacturer_id, model) values (?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatementOfCar = connection.prepareStatement(queryForCar,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatementOfCar.setLong(1, car.getManufacturer().getId());
+            preparedStatementOfCar.setString(2, car.getModel());
+            preparedStatementOfCar.executeUpdate();
+
+            ResultSet generatedKeys = preparedStatementOfCar.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                car.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't create the data - "
+                    + car + " into cars", exception);
+        }
         return car;
     }
 
     @Override
     public Optional<Car> get(Long id) {
-        return Storage.getCars().stream()
-                .filter(x -> x.getId().equals(id))
-                .findFirst();
+        Optional<Car> optionalCar;
+        String queryCreateCarAndManufacturer = "select cars.id "
+                + "as c_id, cars.manufacturer_id as c_mid, "
+                + "cars.model as c_m, manufacturers.id as m_id, manufacturers.name as m_n, "
+                + "manufacturers.country as m_c from cars "
+                + "inner join manufacturers on manufacturers.id = cars.manufacturer_id "
+                + "where cars.deleted = false and manufacturers.deleted = false and cars.id = ?;";
+        String queryCreateDriver = "select cars.id as c_id, drivers.id as d_id, "
+                + "drivers.name as d_n, drivers.license_number as d_ln from cars "
+                + "inner join cars_drivers on cars_drivers.car_id = cars.id "
+                + "inner join drivers on drivers.id = cars_drivers.driver_id "
+                + "where cars.id = ? and drivers.deleted = false;";
```

Comment:
```suggestion
        String query = "select id, "
                + "name, license_number from drivers d "
                + "inner join cars_drivers cd on cd.driver_id = d.id "
                + "where car_id = ? and deleted = false;";
```
the query in private method `getDriversByCarId()` should look like this

Reply:
okey, i created that method